src/OrleansProviders/Streams/Memory/MemoryStreamProvider.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Orleans.Providers.Streams.Common;
+
+namespace Orleans.Providers.Streams.Memory
+{
+    /// <summary>
+    /// This is a persistent stream provider that uses in-memory grain to queue the events.
+    /// This is primarily for test purposes.
```

Comment:
Would not an easier and actually better approach is to build an in-memory emulation of AzureQueueManager. It will expose the same set of low level APIs as AzureQueueManager. It will require significantly less new code, as now you are mostly duplicating the other providers.
If you want to model it after EH, you could do the same - in memory EH emulation.
Then you could just slightly refactor the existing EH or AQ to inject the queue manager.